internal/dcache_lib/rpc/client/connection.go

```diff
@@ -0,0 +1,155 @@
+/*
+    _____           _____   _____   ____          ______  _____  ------
+   |     |  |      |     | |     | |     |     | |       |            |
+   |     |  |      |     | |     | |     |     | |       |            |
+   | --- |  |      |     | |-----| |---- |     | |-----| |-----  ------
+   |     |  |      |     | |     | |     |     |       | |       |
+   | ____|  |_____ | ____| | ____| |     |_____|  _____| |_____  |_____
+
+
+   Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+
+   Copyright © 2020-2025 Microsoft Corporation. All rights reserved.
+   Author : <blobfusedev@microsoft.com>
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE
+*/
+
+package client
+
+import (
+	"context"
+	"crypto/tls"
+
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/internal/dcache_lib/rpc/gen-go/dcache/models"
+	"github.com/Azure/azure-storage-fuse/v2/internal/dcache_lib/rpc/gen-go/dcache/service"
+	"github.com/apache/thrift/lib/go/thrift"
+)
+
+// Connection struct holds the Thrift connection to a node
+// This is used to make RPC calls to the node
+type Connection struct {
+	nodeID      string                      // Node ID of the node this connection is for, can be used for debug logs
+	nodeAddress string                      // Address of the node this connection is for
+	ctx         context.Context             // Context for the connection
+	Transport   thrift.TTransport           // Transport is the Thrift transport layer
+	Client      *service.ChunkServiceClient // Client is the Thrift client for the ChunkService
+}
+
```

Comment:
this should be called rpcClient as it is indeed an RPC client using which we can call RPC methods. It has the following properties:
- it's connected to an RPC server (the nodeid/nodeaddress)
- it has a ref to the thrift transport needed to carry the requests
- it has a ref to the ServiceClient needed to call thrift RPCs

External users of this package need not be aware of this, so no need to export it.
The only exported things from this package should be the RPC methods.

Reply:
Done